tests/ZfrCorsTest/Service/CorsServiceTest.php

```diff
@@ -236,6 +236,23 @@ public function testReturnNullForUnknownOrigin()
         $this->assertEquals('null', $headers->get('Access-Control-Allow-Origin')->getFieldValue());
     }
 
+    public function testEnsureVaryHeaderForNoOrigin()
+    {
+        $request  = new HttpRequest();
+        $response = new HttpResponse();
+
+        if (!$this->corsService->isCorsRequest($request)) {
+            if (!$request->getHeader("Origin")) {
+                $this->corsService->ensureVaryHeader($response);
```

Comment:
I don't think you need all those conditions, you already know if the test that it's not a cors request and that it does not have the origin header, so just call "ensureVaryHeader".